.github/copilot-instructions.md

```diff
@@ -0,0 +1,326 @@
+```bash
+# On Ubuntu (requires Intel apt repository setup)
+sudo apt install intel-oneapi-mkl intel-oneapi-mkl-devel
+source /opt/intel/oneapi/setvars.sh
+```
+
+### Common Build Options (from cmake/options.cmake)
+
+| Option | Default | Description |
+|--------|---------|-------------|
+| `SVS_BUILD_BINARIES` | OFF | Build utility binaries in utils/ |
+| `SVS_BUILD_TESTS` | OFF | Build test suite (Catch2-based) |
+| `SVS_BUILD_EXAMPLES` | OFF | Build C++ examples |
+| `SVS_BUILD_BENCHMARK` | OFF | Build benchmark executable |
+| `SVS_NO_AVX512` | OFF | Disable Intel AVX-512 intrinsics |
+| `SVS_EXPERIMENTAL_ENABLE_IVF` | OFF | Enable IVF (requires MKL) |
+| `CMAKE_BUILD_TYPE` | Release | Use `RelWithDebugInfo` for testing |
+
+## Code Formatting and Linting
+
+### Formatting (ALWAYS run before committing)
+
+**Tool**: clang-format version 15.x (specified in `.pre-commit-config.yaml`)
+- **DO NOT** use clang-format 16+ or 14 and below - version 15.x is required
+
+```bash
+# Format all code (run from repository root)
+./tools/clang-format.sh clang-format
+
+# Formatted directories: bindings/python/src, bindings/python/include, 
+#                        include, benchmark, tests, utils, examples/cpp
+```
+
+### Pre-commit Hooks
+
+The repository uses pre-commit for automated formatting checks:
+
+```bash
+# Install pre-commit (if not already installed)
+pip install pre-commit
+
+# Install hooks (one-time setup, takes 1-2 minutes)
+pre-commit install-hooks
+
+# Run manually (optional, CI will check)
+pre-commit run --all-files
+```
+
+**CI check**: The `pre-commit.yml` workflow runs on all PRs and will fail if code is not formatted.
+
+## Testing
+
+### C++ Tests (Catch2)
+
+Tests use Catch2 v3 with prefix macros (`CATCH_TEST_CASE`, `CATCH_REQUIRE`, etc.):
+
+```bash
+# From build/tests directory
+cd build/tests
+
+# Run all tests
+ctest -C RelWithDebugInfo
+# OR
+./tests
+
+# Run specific test tags
+./tests "[integration][build]"
+./tests "[core][distance]"
+
+# List available tags
+./tests --list-tags
+
+# Run with verbose output
+CTEST_OUTPUT_ON_FAILURE=1 ctest -C RelWithDebugInfo
+```
+
+**Test tags commonly used**: `[integration]`, `[build]`, `[core]`, `[distance]`, `[vamana]`, `[data]`
+
+### C++ Examples
+
+Examples are tested via ctest:
+
+```bash
+cd build/examples/cpp
+ctest -C RelWithDebugInfo
+# Runs 10 example tests (~9 seconds total)
+```
+
+### Python Tests
+
+Python tests use pytest (location: `bindings/python/tests/`):
+
+```bash
+# Build Python bindings first (requires scikit-build)
+cd bindings/python
+pip install -e .
+
+# Run tests
+pytest tests/
+```
+
+## Project Structure
+
+```
+ScalableVectorSearch/
+├── .github/
+│   ├── workflows/           # CI/CD pipelines
+│   │   ├── build-linux.yml  # Main build & test (Ubuntu 22.04, g++/clang)
+│   │   ├── pre-commit.yml   # Format checking
+│   │   ├── cibuildwheel.yml # Python wheel building
+│   │   └── build-*.y{a}ml   # macOS, ARM builds
```

Comment:
Didn't notice there was inconsistencies here. Maybe makes sense to change to .yml and revise file name (build-macos.yml) in this PR.